From a6ede77250a4406ac71cf41ea46743af857bbe0f Mon Sep 17 00:00:00 2001
From: Yuri Tolstov <yuri.tolstov@cavium.com>
Date: Tue, 21 Nov 2017 15:15:22 -0800
Subject: [PATCH 0356/1051] octeontx-sso: Turn off SSO conditional clocking

CN83xx errata SSO-29000
With conditional clock logic being turned on, the issue exists of losing
internal credits during add-work operation, which eventually leads to SSO
to hang.

Signed-off-by: Yuri Tolstov <yuri.tolstov@cavium.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c
index 4d2b7d015cca..e2537e458261 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c
@@ -1130,6 +1130,10 @@ static inline void sso_configure_on_chip_res(struct ssopf *sso, u16 ssogrps)
 			      (taq_rsvd_cnt & SSO_TAQ_CNT_RSVD_CNT_MASK) <<
 				SSO_TAQ_CNT_RSVD_CNT_SHIFT);
 	}
+	/* Turn off SSO conditional clocking (Errata SSO-29000) */
+	tmp = sso_reg_read(sso, SSO_PF_WS_CFG);
+	tmp |= 0x1; /*SSO_CCLK_DIS*/
+	sso_reg_write(sso, SSO_PF_WS_CFG, tmp);
 }
 
 static inline void sso_max_grps_update(struct ssopf *sso)
-- 
2.17.1

